spyder/plugins/completion/plugin.py

```diff
@@ -139,8 +139,16 @@ def _maybe_send_locked(self, req_id):
         request_responses = self.requests[req_id]
 
         def send():
+            max_req_id = max(
```

Comment:
Please add a comment above this line referencing the issue (just to keep some traceability). Something like:

```python
# Needed to prevent the send of completions for old requests
# See spyder-ide/spyder#10798
```